Mage.Sets/src/mage/cards/j/JayaBallard.java

```diff
@@ -0,0 +1,237 @@
+
+        addSuperType(SuperType.LEGENDARY);
+        this.subtype.add(SubType.JAYA);
+
+        this.addAbility(new PlanswalkerEntersWithLoyalityCountersAbility(5));
+
+        // +1: Add {R}{R}{R}. Spend this mana only to cast instant or sorcery spells.
+        this.addAbility(
+            new LoyaltyAbility(
+                new AddConditionalManaEffect(
+                    Mana.RedMana(3),
+                    new ConditionalSpellManaBuilder(
+                        new FilterInstantOrSorcerySpell())),
+                1));
+
+        // +1: Discard up to three cards, then draw that many cards.
+        this.addAbility(
+            new LoyaltyAbility(
+                new DiscardUpToNDrawThatManySourceEffect(3),
+                1));
+        // −8: You get an emblem with "You may cast instant and sorcery cards from your graveyard. If a card cast this way would be put into your graveyard, exile it instead."
+        this.addAbility(
+            new LoyaltyAbility(
+                new GetEmblemEffect(new JayaBallardEmblem()),
+                -8));
+    }
+
+    public JayaBallard(final JayaBallard card) {
+        super(card);
+    }
+
+    @Override
+    public JayaBallard copy() {
+        return new JayaBallard(this);
+    }
+}
+
+class JayaBallardEmblem extends Emblem {
+    // You may cast instant or sorcery cards from your graveyard. If a card cast this way would be put into your graveyard, exile it instead
+
+    public JayaBallardEmblem() {
+        this.setName("Emblem Jaya");
+        this.getAbilities().add(new SimpleStaticAbility(Zone.COMMAND, new JayaBallardEmblemContinuousEffect()));
+    }
+}
+
+class JayaBallardEmblemContinuousEffect extends ContinuousEffectImpl {
+
+    private static final FilterCard filter = new FilterCard("Instant or sorcery spell");
+
+    static {
+        filter.add(Predicates.or(
+            new CardTypePredicate(CardType.INSTANT),
+            new CardTypePredicate(CardType.SORCERY)
+        ));
+    }
+
+    JayaBallardEmblemContinuousEffect() {
+        super(Duration.WhileOnBattlefield, Layer.PlayerEffects, SubLayer.NA, Outcome.Benefit);
+        staticText = "You may cast instant or sorcery cards from your graveyard. If a card cast this way would be put into your graveyard, exile it instead";
+    }
+
+    JayaBallardEmblemContinuousEffect(final JayaBallardEmblemContinuousEffect effect) {
+        super(effect);
+    }
+
+    @Override
+    public JayaBallardEmblemContinuousEffect copy() {
+        return new JayaBallardEmblemContinuousEffect(this);
+    }
+
+    @Override
+    public boolean apply(Game game, Ability source) {
+        Player player = game.getPlayer(source.getControllerId());
+        if (player == null) {
+            return false;
+        }
+        for (Card card : player.getGraveyard().getCards(filter, game)) {              
+            ContinuousEffect effect = new JayaBallardEmblemCastFromGraveyardEffect();
```

Comment:
This enters the same continuous effects over an over again for each applyEffects call.
I guess you need only two continous effects that do the same for all cards in the controllers graveyard.

Reply:
Thank you, I will have to try a different route with this one. I re-implemented the emblem as a simple static ability, but I don't know how to tie the replacement effect to that ability. My original code was based on kess, dissident mage, which has the same bug.